Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8024,21 +8024,30 @@ void WebPage::updateAttachmentIcon(const String& identifier, ShareableBitmap::Ha
                     icon->paint(imageBuffer->context(), IntPoint::zero(), IntRect(IntPoint::zero(), icon->size()));
                     auto data = imageBuffer->toData("image/png"_s);
                     attachment->updateIconForWideLayout(WTFMove(data));
+                    return;
                 }
-            } else
+            } else {
                 attachment->updateIconForNarrowLayout(icon->createImage(), size);
+                return;
+            }
         }
+
+        if (attachment->isWideLayout())
+            attachment->updateIconForWideLayout({ });
+        else
+            attachment->updateIconForNarrowLayout({ }, size);
     }
 }
 
-void WebPage::requestAttachmentIcon(const String& identifier, const WebCore::FloatSize& size)
+bool WebPage::requestAttachmentIcon(const String& identifier, const WebCore::FloatSize& size)
 {
     if (auto attachment = attachmentElementWithIdentifier(identifier)) {
         String fileName;
         if (File* file = attachment->file())
             fileName = file->path();
-        send(Messages::WebPageProxy::RequestAttachmentIcon(identifier, attachment->attachmentType(), fileName, attachment->attachmentTitle(), size));
+        return send(Messages::WebPageProxy::RequestAttachmentIcon(identifier, attachment->attachmentType(), fileName, attachment->attachmentTitle(), size));
```

Comment:
I don't think we normally use the return value of `send`. I would just return `true` after calling it.

Reply:
Well, the goal was to know if the message was actually sent, so the caller `HTMLAttachmentElement::requestWideLayoutIconIfNeeded()` knows whether to set its `m_requestingIcon`: If the message was not even sent, then we're not waiting for an icon.

Reply:
I guess it's fine. Just not something we normally do, as something has gone terribly wrong if `send` fails.

Reply:
I've now removed that code.